tests/parser/types/test_lists.py

```diff
@@ -428,6 +428,45 @@ def ix(i: uint256) -> {type}:
     assert_tx_failed(lambda: c.ix(len(value) + 1))
 
 
+def test_constant_list_fail(get_contract, assert_compile_failed, distinct_int_types):
+    type = distinct_int_types[0]
+    return_type = distinct_int_types[1]
+    code = f"""
+MY_CONSTANT: constant({type}[3]) = [1, 2, 3]
+
+@external
+def foo() -> {return_type}[3]:
+    return MY_CONSTANT
+    """
+    assert_compile_failed(lambda: get_contract(code), InvalidType)
+
+
+def test_constant_list_fail_2(get_contract, assert_compile_failed, distinct_int_types):
+    type = distinct_int_types[0]
+    return_type = distinct_int_types[1]
```

Comment:
Side note: Python supports ["tuple destructuring"](https://blog.teclado.com/destructuring-in-python/)
```suggestion
    type, return_type = distinct_int_types
```